src/sas/mod.rs

```diff
@@ -0,0 +1,190 @@
+    I = b'I',
+    /// A missing value represented using the ASCII letter 'J'.
+    J = b'J',
+    /// A missing value represented using the ASCII letter 'K'.
+    K = b'K',
+    /// A missing value represented using the ASCII letter 'L'.
+    L = b'L',
+    /// A missing value represented using the ASCII letter 'M'.
+    M = b'M',
+    /// A missing value represented using the ASCII letter 'N'.
+    N = b'N',
+    /// A missing value represented using the ASCII letter 'O'.
+    O = b'O',
+    /// A missing value represented using the ASCII letter 'P'.
+    P = b'P',
+    /// A missing value represented using the ASCII letter 'Q'.
+    Q = b'Q',
+    /// A missing value represented using the ASCII letter 'R'.
+    R = b'R',
+    /// A missing value represented using the ASCII letter 'S'.
+    S = b'S',
+    /// A missing value represented using the ASCII letter 'T'.
+    T = b'T',
+    /// A missing value represented using the ASCII letter 'U'.
+    U = b'U',
+    /// A missing value represented using the ASCII letter 'V'.
+    V = b'V',
+    /// A missing value represented using the ASCII letter 'W'.
+    W = b'W',
+    /// A missing value represented using the ASCII letter 'X'.
+    X = b'X',
+    /// A missing value represented using the ASCII letter 'Y'.
+    Y = b'Y',
+    /// A missing value represented using the ASCII letter 'Z'.
+    Z = b'Z',
+}
+
+impl MissingValue {
+    /// Return the ASCII code representation for the missing value.
+    #[inline]
+    pub const fn code(&self) -> u8 {
+        *self as u8
+    }
+}
+
+impl TryFrom<u8> for MissingValue {
+    type Error = &'static str;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            b'.' => Ok(MissingValue::Period),
+            b'_' => Ok(MissingValue::Underscore),
+            b'A' => Ok(MissingValue::A),
+            b'B' => Ok(MissingValue::B),
+            b'C' => Ok(MissingValue::C),
+            b'D' => Ok(MissingValue::D),
+            b'E' => Ok(MissingValue::E),
+            b'F' => Ok(MissingValue::F),
+            b'G' => Ok(MissingValue::G),
+            b'H' => Ok(MissingValue::H),
+            b'I' => Ok(MissingValue::I),
+            b'J' => Ok(MissingValue::J),
+            b'K' => Ok(MissingValue::K),
+            b'L' => Ok(MissingValue::L),
+            b'M' => Ok(MissingValue::M),
+            b'N' => Ok(MissingValue::N),
+            b'O' => Ok(MissingValue::O),
+            b'P' => Ok(MissingValue::P),
+            b'Q' => Ok(MissingValue::Q),
+            b'R' => Ok(MissingValue::R),
+            b'S' => Ok(MissingValue::S),
+            b'T' => Ok(MissingValue::T),
+            b'U' => Ok(MissingValue::U),
+            b'V' => Ok(MissingValue::V),
+            b'W' => Ok(MissingValue::W),
+            b'X' => Ok(MissingValue::X),
+            b'Y' => Ok(MissingValue::Y),
+            b'Z' => Ok(MissingValue::Z),
+            _ => Err("Not a missing value"),
+        }
```

Comment:
I was surprised I needed to create a separate case for every value. I assume I could just cast the `u8` into a `MissingValue`, but not without unsafe code. I wasn't sure if `TryFrom` was the right approach here.